modules/diskq/Makefile.am

```diff
@@ -22,15 +14,43 @@ modules_diskq_libdisk_buffer_la_SOURCES = \
   modules/diskq/qdisk.h \
   modules/diskq/qdisk.c
 
-modules_diskq_libdisk_buffer_la_CFLAGS = $(AM_CPPFLAGS) -I$(top_srcdir)/modules/diskq -I$(top_builddir)/modules/diskq -I$(top_srcdir)/lib -I../../lib
-modules_diskq_libdisk_buffer_la_LIBADD = $(MODULE_DEPS_LIBS)
+modules_diskq_libsyslog_ng_disk_buffer_la_CPPFLAGS = \
+  $(AM_CPPFLAGS) \
+  -I$(top_srcdir)/modules/diskq
+modules_diskq_libsyslog_ng_disk_buffer_la_LIBADD	=	\
+	$(MODULE_DEPS_LIBS)
+modules_diskq_libsyslog_ng_disk_buffer_la_DEPENDENCIES	=	\
+	$(MODULE_DEPS_LIBS)
+
+modules_diskq_libdisk_buffer_la_SOURCES = \
+  modules/diskq/diskq.c \
+  modules/diskq/diskq.h \
+  modules/diskq/diskq-grammar.y \
+  modules/diskq/diskq-parser.c \
+  modules/diskq/diskq-parser.h \
+  modules/diskq/diskq-plugin.c
+
+BUILT_SOURCES += \
+  modules/diskq/diskq-grammar.y \
+  modules/diskq/diskq-grammar.c \
+  modules/diskq/diskq-grammar.h
+
+EXTRA_DIST += modules/diskq/diskq-grammar.ym
+
+modules_diskq_libdisk_buffer_la_CPPFLAGS = $(AM_CPPFLAGS) -I$(top_srcdir)/modules/diskq -I$(top_builddir)/modules/diskq
+modules_diskq_libdisk_buffer_la_LIBADD = $(MODULE_DEPS_LIBS) $(top_builddir)/modules/diskq/libsyslog-ng-disk-buffer.la
```

Comment:
You could consider lifting this into a new variable: `$(top_builddir)/modules/diskq/libsyslog-ng-disk-buffer.la`